python/lsst/ts/ofc/sensitivity_matrix.py

```diff
@@ -0,0 +1,152 @@
+import galsim
+import lsst.obs.lsst as obs_lsst
+from lsst.afw.cameraGeom import FIELD_ANGLE
+
+
+class SensitivityMatrix:
+    """Class to handle the sensitivity matrix.
+
+    Parameters
+    ----------
+    ofc_data : `lsst.ts.ofc.ofc_data.OFCData`
+        OFC data.
+    """
+
+    def __init__(self, ofc_data) -> None:
+        self.ofc_data = ofc_data
+
+    def evaluate_sensitivity(
+        self, rotation_angle: float, sensor_names: None | list = None
+    ) -> Tuple[npt.NDArray[np.float64], npt.NDArray[np.float64]]:
+        """Evaluate the sensitivity matrix for a given rotation angle.
+
+        Parameters
+        ----------
+        rotation_angle : `float`
+            Rotation angle in degrees.
+        sensor_names : `list` of `str`, optional
+            List of sensor names. If None, uses all the GQ points are used.
+
+        Returns
+        -------
+        `np.array` of `float`
+            Sensitivity matrix for the given rotation angle.
+        `np.array` of `int`
+            Field indices for the sensors.
+        """
+
+        # If sensor_names is None getsensitivity matrix at all GQ points
+        if sensor_names is None:
+            # Create array for all the GQ points
+            field_idx = np.arange(self.ofc_data.gq_field_angles.shape[0])
+            # Retrieve field angles for each GQ point from ofc_data
+            field_x, field_y = zip(*self.ofc_data.gq_field_angles)
+
+        else:
+            # Obtain the field indices for the sensors
+            field_idx = np.array(
+                [self.ofc_data.field_idx[sensor_name] for sensor_name in sensor_names]
+            )
+            # Get the field angles for the sensors
+            field_x, field_y = get_field_angle(sensor_names)
+
+        # If not using double zernikes, use the stored GQ sensitivity matrix
+        if not self.ofc_data.double_zernikes:
+            # Get the sensitivity matrix
+            sen_m = self.ofc_data.sensitivity_matrix[
+                np.ix_(
+                    np.arange(self.ofc_data.sensitivity_matrix.shape[0]),
+                    np.arange(self.ofc_data.sensitivity_matrix.shape[1]),
+                    self.ofc_data.dof_idx,
+                )
+            ]
+
+            # return the sensitivity matrix at the given field indices.
+            return sen_m[field_idx, :, :], field_idx
+
+        # If using double zernikes, compute the sensitvity mat
+        # using galsim DoubleZernikes
+        rotated_sensitivity_matrix = np.array(
+            [
+                np.array(
+                    [
+                        zk.coef
+                        for zk in galsim.zernike.DoubleZernike(
+                            self.ofc_data.sensitivity_matrix[:, :, dof_idx],
+                            # Rubin annuli
+                            uv_inner=0.0,
+                            uv_outer=1.75,
+                            xy_inner=0.612 * 4.18,
+                            xy_outer=4.18,
```

Comment:
These parameters should be in the configuration file:

1. https://github.com/lsst-ts/ts_wep/blob/develop/policy/cwfs/instData/lsst/instParamPipeConfig.yaml
2. https://github.com/lsst-ts/ts_wep/blob/develop/policy/cwfs/instData/comcam/instParamPipeConfig.yaml
3. https://github.com/lsst-ts/ts_wep/blob/develop/policy/cwfs/instData/lsstfam/instParamPipeConfig.yaml